### PR TITLE
Created the Median seroprevalence by WHO region and age group table

### LINKED ISSUE #82: Improvements to the "Median seroprevalence by WHO region and age group" graph
The "Median seroprevalence by WHO region and age group" should have a legend or shouldn't be a graph at all and should be in tabular format because we found the graph busy and difficult to read.

### PR DESCRIPTION
**Pull request by SeanKennyNF**
Resolves #82. Created the Median seroprevalence by WHO region and age group table

Peek 2024-03-23 20-31
image
median-seroprevalence-by-who-region-and-age-group
image
median-seroprevalence-by-who-region-and-age-group(1)

### CODE REVIEW

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+} from "@/components/ui/table";
+import { WHORegion } from "@/lib/who-regions";
+import { Button } from "@/components/ui/button";
+
+enum AgeGroup {
+  "Adults (18-64 years)" = "Adults (18-64 years)",
+  "Children and Youth (0-17 years)" = "Children and Youth (0-17 years)",
+  "Seniors (65+ years)" = "Seniors (65+ years)",
+  "Multiple groups" = "Multiple groups",
+}
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:28]: Yeah I know there's a type in `recharts.tsx`, it's just I can do an `Object.values` on a union type.

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    "ZIKV",
+    "DENV",
+    "CHIKV",
+    "YF",
+    "WNV",
+    "MAYV",
+  ];
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const datasetGroupedByArbovirus = useMemo(
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:42]: Yeah I don't really like what I'm doing with the grouping functions either. I want to make a more generalized triple group function like we have with the bar charts double grouping function but figured it doesn't really have to happen right now right now.

`useMemo`s for performance ofc

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  );
+
+  const datasetToDisplay = useMemo(
+    () => tableDatasets[currentlySelectedArbovirus],
+    [currentlySelectedArbovirus, tableDatasets]
+  );
+
+  return (
+    <div className="p-2">
+      <div className="flex justify-between mb-2 ignore-for-visualization-download">
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:102]: Don't want the header and the buttons to get included in the downloaded png. I'll swap over the buttons to use this same method at some point, not this PR though.

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                        oneValuedColourHexCode: "#f79489",
+                        value: seroprevalenceNumber / 100,
+                      })
+                    : "#ffffff";
+
+                return (
+                  <TableCell
+                    className="border-l border-r border-b bg-white group-hover:bg-zinc-100 whitespace-nowrap"
+                    key={`median-seroprevalence-by-who-region-and-age-group-table-${whoRegion}-${ageGroup}-cell`}
+                    style={{ backgroundColor: backgroundColourHexCode }}
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:179]: Can't do this in tailwind ofc

File: components/ui/data-table/data-table.tsx [hunk 5]
```diff
@@ -231,6 +231,7 @@ export function DataTable<TData, TValue>(props: DataTableProps<TData, TValue>) {
           <Button
             variant="outline"
             size="sm"
+            className="text-white"
```
Comment (SeanKennyNF) [inline components/ui/data-table/data-table.tsx:234]: Black text isn't legible on the arbo dark green.

File: app/pathogen/arbovirus/dashboard/(visualizations)/visualization-header.tsx [hunk 6]
```diff
@@ -60,6 +60,7 @@ const DownloadButton = (props: DownloadButtonProps) => (
 
 const CloseButton = (props: CloseButtonProps) => (
   <button
+    id={props.configuration.id}
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/visualization-header.tsx:63]: whoops lmao

**Review by EliasPinno — APPROVED** [review]

**Review by EliasPinno — APPROVED** [review]

**Pull request merged**